scripts/create_irreducible_polys_database.py

```diff
@@ -0,0 +1,101 @@
+"""
+A script to create a database of irreducible polynomials
+
+Sources:
+ - Gadiel Seroussi. Table of Low-Weight Binary Irreducible Polynomials (1998): https://www.hpl.hp.com/techreports/98/HPL-98-135.html
+
+"""
+from __future__ import annotations
+
+import os
+import sqlite3
+from pathlib import Path
+
+from pdfminer.high_level import extract_text
+
+
+def main():
+    """
+    The main routine to create a database of irreducible polynomials
+    """
+
+    database_file = Path(__file__).parent.parent / "src" / "galois" / "_databases" / "irreducible_polys.db"
+    conn, cursor = create_database(database_file)
+
+    _add_hpl_1998(conn, cursor)
+
+    conn.close()
+
+
+def create_database(file: Path) -> tuple[sqlite3.Connection, sqlite3.Cursor]:
+    """
+    Deletes the old database, makes a new one, and returns the database connection.
+    """
+    if file.exists():
+        os.remove(file)
+
+    conn = sqlite3.connect(file)
+    cursor = conn.cursor()
+    create_table(conn, cursor)
+
+    return conn, cursor
+
+
+def create_table(conn: sqlite3.Connection, cursor: sqlite3.Cursor):
+    """
+    Creates an empty 'polys' table.
+    """
+    cursor.execute(
+        """
+        CREATE TABLE polys (
+            characteristic INTEGER NOT NULL,
+            degree INTEGER NOT NULL,
+            nonzero_degrees TEXT NOT NULL,
+            nonzero_coeffs TEXT NOT NULL,
+            PRIMARY KEY (characteristic, degree)
+        )
+        """
+    )
+    conn.commit()
+
+
+def add_to_database(cursor: sqlite3.Cursor, characteristic: str, degree: str, nonzero_degrees: str, nonzero_coeffs: str):
```

Comment:
Any reason to keep `characteristic` and `degree` as `str` instead of `int`?

Reply:
No. I agree that they should be `int`.

Reply:
Changed.